src/recipient-recieved-scanner.ts

```diff
@@ -261,12 +263,27 @@ export class RecipientRecievedScanner {
         endBlock,
       );
 
-      // Collect all events without parsing (out of scope for this issue)
-      void events; // Events collected but not processed yet
+      // Parse and store events (will be accumulated)
+      if (events.length > 0) {
+        await this.parseAndStoreEvents(address, events);
```

Comment:
Instead of storing now as we go we should accumulate this data and store once at the end